dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -54,679 +44,240 @@
+   * @param languageId The language ID used to created the Vanity URLs.
+   * @param includeSystemHost If set to {@code true}, the results will include Vanity URLs that were
+   *        created under System Host. Otherwise, set to {@code false}.
+   *
+   * @return The list of Vanity URLs.
+   */
+  @Override
+  @CloseDBIfOpened
+  public List<CachedVanityUrl> findInDb(final Host host, Language lang) {
+
+    try {
+      final List<Map<String, Object>> vanityUrls =
+          new DotConnect().setSQL(SELECT_LIVE_VANITY_URL_INODES).addParam(host.getIdentifier()).addParam(lang.getId()).loadObjectResults();
+      final List<String> vanityUrlInodes =
+          vanityUrls.stream().map(vanity -> vanity.get("live_inode").toString()).collect(Collectors.toList());
+      final List<Contentlet> contentlets = this.contentletAPI.findContentlets(vanityUrlInodes);
+
+      return contentlets.stream().map(c -> new CachedVanityUrl(this.fromContentlet(c))).collect(Collectors.toList());
+
+    } catch (final Exception e) {
+      Logger.error(this,
+          String.format("An error occurred when retrieving Vanity URLs: siteId=[%s], " + "languageId=[%s], includeSystemHost=[%s]",
+              host.getIdentifier(), lang.getId()),
+          e);
+      throw new DotStateException(e);
     }
 
-    @CloseDBIfOpened
-    @Override
-    public boolean isVanityUrl(final String url, final Host host, final long languageId) {
-
-        final String uri    = url; // not process now for it.
-        final String siteId = this.getSiteId(host);
-
-        // tries first cache.
-        final CachedVanityUrl cachedVanityUrl =
-                this.vanityUrlServices
-                        .getCachedVanityUrlByUri(uri, siteId, languageId);
-        boolean isVanityURL = !is404(cachedVanityUrl)
-                && this.patternMatches(cachedVanityUrl, uri).isPatternMatches();
-
-        if (!isVanityURL) {
-
-            CachedVanityUrl liveCachedVanityUrl =
-                    this.searchLiveCachedVanityUrlBySiteAndLanguage
-                            (uri, siteId, languageId);
-
-            if (null == liveCachedVanityUrl) {
-
-                liveCachedVanityUrl =
-                        this.getFallback(uri, siteId, languageId, this.systemUser);
-            }
-
-            isVanityURL =
-                    !is404(liveCachedVanityUrl);
+  }
 
-            // Still support legacy cmsHomePage
-            if (URL_SUFFIX.equals(uri) && !isVanityURL) {
 
-                liveCachedVanityUrl =
-                        this.getLiveCachedVanityUrl
-                                (LEGACY_CMS_HOME_PAGE, host, languageId, this.systemUser);
-                isVanityURL = UtilMethods.isSet(liveCachedVanityUrl)
-                        && !VanityUrlAPI.CACHE_404_VANITY_URL
-                            .equals(liveCachedVanityUrl.getVanityUrlId());
-            }
+  
+  
+  
+  private List<CachedVanityUrl> load(Host host, Language lang){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'lang' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)